src/routes/(app)/news/ArticleForm.svelte

```diff
@@ -52,6 +53,31 @@
       }
     };
   };
+
+  const onVideoFileSelected = (
+    event: Event & {
+      currentTarget: EventTarget & HTMLInputElement;
+    },
+  ) => {
+    let video = event.currentTarget.files?.[0];
+    if (!video) return;
+    let reader = new FileReader();
+    reader.readAsDataURL(video);
+    reader.onload = (e) => {
+      let result = e.target?.result ?? undefined;
+      // If array buffer, convert to base64
+      if (result instanceof ArrayBuffer) {
+        articleVideo = btoa(
+          new Uint8Array(result).reduce(
+            (data, byte) => data + String.fromCharCode(byte),
+            "",
+          ),
+        );
+      } else {
+        articleVideo = result;
+      }
+    };
+  };
```

Comment:
I'm not really sure how this code works 😅, but I have a few thoughts:
1. For one, it's very similar to the code for handling images. Maybe it could just be one function?
2. Does it work for large videos? As far as I can tell, the function converts the video to a long [base64](https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data) string and that feels a bit weird to me - is there any limit to how long a base64 string can be?